spec.html

```diff
@@ -0,0 +1,202 @@
+      <h1>GetExportedNames( _exportStarSet_ )</h1>
+
+      <p>The GetExportedNames concrete method of a Synthetic Module Record implements the corresponding Module Record abstract method.</p>
+      <p>It performs the following steps:</p>
+
+      <emu-alg>
+        1. Let _module_ be this Synthetic Module Record.
+        1. Return _module_.[[ExportNames]].
+      </emu-alg>
+    </emu-clause>
+
+    <emu-clause id="sec-smr-resolveexport">
+      <h1>ResolveExport( _exportName_, _resolveSet_ )</h1>
+
+      <p>The ResolveExport concrete method of a Synthetic Module Record implements the corresponding Module Record abstract method.</p>
+      <p>It performs the following steps:</p>
+
+      <emu-alg>
+        1. Let _module_ be this Synthetic Module Record.
+        1. If _module_.[[ExportNames]] does not contain _exportName_, return null.
+        1. Return ResolvedBinding Record { [[Module]]: _module_, [[BindingName]]: _exportName_ }.
+      </emu-alg>
+    </emu-clause>
+
+    <emu-clause id="sec-smr-instantiate">
+      <h1>Instantiate ( )</h1>
+
+      <p>The Instantiate concrete method of a Synthetic Module Record implements the corresponding Module Record abstract method.</p>
+      <p>It performs the following steps:</p>
+
+      <emu-alg>
+        1. Let _module_ be this Synthetic Module Record.
+        1. Let _realm_ be _module_.[[Realm]].
+        1. Assert: _realm_ is not *undefined*.
+        1. Let _env_ be NewModuleEnvironment(_realm_.[[GlobalEnv]]).
+        1. Set _module_.[[Environment]] to _env_.
+        1. Let _envRec_ be _env_'s EnvironmentRecord.
+        1. For each _exportName_ in _module_.[[ExportNames]],
+          1. Perform ! _envRec_.CreateMutableBinding(_exportName_, *false*).
+          1. Perform ! _envRec_.InitializeBinding(_exportName_, *undefined*).
+        1. Return *undefined*.
+      </emu-alg>
+    </emu-clause>
+
+    <emu-clause id="sec-smr-Evaluate">
+      <h1>Evaluate ( )</h1>
+
+      <p>The Evaluate concrete method of a Synthetic Module Record implements the corresponding Module Record abstract method.</p>
+      <p>It performs the following steps:</p>
+
+      <emu-alg>
+        1. Let _module_ be this Synthetic Module Record.
+        1. Let _moduleCxt_ be a new ECMAScript code execution context.
+        1. Set the Function of _moduleCxt_ to *null*.
+        1. Assert: _module_.[[Realm]] is not *undefined*.
+        1. Set the Realm of _moduleCxt_ to _module_.[[Realm]].
+        1. Set the ScriptOrModule of _moduleCxt_ to _module_.
+        1. Set the VariableEnvironment of _moduleCxt_ to _module_.[[Environment]].
+        1. Set the LexicalEnvironment of _moduleCxt_ to _module_.[[Environment]].
+        1. Suspend the currently running execution context.
+        1. Push _moduleCxt_ on to the execution context stack; _moduleCxt_ is now the running execution context.
+        1. Let _result_ be the result of performing ? _module_.[[EvaluationSteps]](_module_).
```

Comment:
Should this `?` be here? It seems like the execution context stack will be messed up if this throws.

Reply:
`?` seems right to me--the module evaluation algorithms are all written around `Evaluate()` maybe throwing. See https://github.com/tc39/ecma262/pull/916 for more context.

Reply:
Or, step 5 of https://tc39.github.io/ecma262/#sec-moduleevaluation

Reply:
I think @Ms2ger is right. The ? is wrong; the exception gets re-thrown (or the normal completion gets returned) 3 steps down, but before doing that we need to clean up the execution context stack stuff.